src/wallet/rpcwallet.cpp

```diff
@@ -3775,9 +3775,20 @@ static UniValue getaddressesbylabel(const JSONRPCRequest& request)
 
     // Find all addresses that have the given label
     UniValue ret(UniValue::VOBJ);
-    for (const std::pair<CTxDestination, CAddressBookData>& item : pwallet->mapAddressBook) {
+    std::set<std::string> addresses;
+    for (const std::pair<const CTxDestination, CAddressBookData>& item : pwallet->mapAddressBook) {
```

Comment:
15463: unrelated, should keep `for (const std::pair<CTxDestination, CAddressBookData>& item : pwallet->mapAddressBook) {` here.

NOTE: we should backport missing parts of 13241.

Reply:
agree and updated code.